packages/lesswrong/lib/helpers.js

```diff
@@ -42,22 +42,37 @@ Users.isSubscribedTo = (user, document) => {
   }
 };
 // LESSWRONG version of getting unused slug. Modified to also include "oldSlugs" array
-Utils.getUnusedSlug = function (collection, slug, useOldSlugs = false) {
+Utils.getUnusedSlug = function (collection, slug, useOldSlugs = false, documentId) {
   let suffix = '';
   let index = 0;
-
+  
+  let existingDocuments = getDocumentsBySlug({slug, suffix, useOldSlugs, collection})
   // test if slug is already in use
-  while (!!collection.findOne(useOldSlugs ? {$or: [{slug: slug+suffix},{oldSlugs: slug+suffix}]} : {slug: slug+suffix})) {
-    index++
-    suffix = '-'+index;
+  while (!!existingDocuments?.length) {
+    // Filter out our own document (i.e. don't change the slug if the only conflict is with ourselves)
+    const conflictingDocuments = existingDocuments.filter((doc) => doc._id !== documentId)
+    // If there are other documents we conflict with, change the index and slug, then check again
+    if (!!conflictingDocuments?.length) {
+      index++
+      suffix = '-'+index;
+      existingDocuments = getDocumentsBySlug({slug, suffix, useOldSlugs, collection, documentId})
```

Comment:
Unused parameter (remnant of an earlier iteration I presume)
```suggestion
      existingDocuments = getDocumentsBySlug({slug, suffix, useOldSlugs, collection})
```